tutorials/cloud-run-with-pomerium-for-end-user-access/index.md

```diff
@@ -0,0 +1,190 @@
+
+Substitute `cloudrun.pomerium.com` for your own subdomain and your e-mail domain if appropriate:
+
+    # policy.template.yaml
+    # see https://www.pomerium.com/reference/#policy
+    - from: https://hello.cloudrun.pomerium.com
+      to: ${HELLO_URL}
+      allowed_domains:
+        - gmail.com
+      enable_google_cloud_serverless_authentication: true
+    - from: https://httpbin.cloudrun.pomerium.com
+      to: https://httpbin.org
+      pass_identity_headers: true
+      allowed_domains:
+        - gmail.com
+
+### DNS in `zonefile.txt`
+
+Substitute `cloudrun.pomerium.com` for your own subdomain (`zonefile.txt`):
+
+    ; zonefile.txt
+    *.cloudrun.pomerium.com. 18000 IN CNAME ghs.googlehosted.com.
+
+Alternatively, you can set an equivalent CNAME in your DNS provider.
+
+## Deployment
+
+Ensure that you have set a default project:
+
+    gcloud config set default-project MYTESTPROJECT
+
+Run the following commands to configure and deploy Pomerium:
+
+    #!/bin/bash
+
+    # Install gcloud beta
+    gcloud components install beta
+
+    # Capture current project number
+    PROJECT=$(gcloud projects describe $(gcloud config get-value project) --format='get(projectNumber)')
+
+    # Point a wildcard domain of *.cloudrun.pomerium.com to the Cloud Run front end
+    gcloud dns record-sets import --zone pomerium-io zonefile --zone-file-format
+
+    # Deploy your protected application and associate a DNS name
+    gcloud run deploy hello --image=gcr.io/cloudrun/hello --region us-central1 --platform managed --no-allow-unauthenticated
+    gcloud run services add-iam-policy-binding hello --platform managed --region us-central1 \
+        --member=serviceAccount:${PROJECT}-compute@developer.gserviceaccount.com \
+        --role=roles/run.invoker
+    gcloud beta run domain-mappings --platform managed --region us-central1 create --service=hello --domain hello-direct.cloudrun.pomerium.com
+
+    # Rewrite policy file with unique 'hello' service URL
+    HELLO_URL=$(gcloud run services describe hello --platform managed --region us-central1 --format 'value(status.address.url)') envsubst <policy.template.yaml >policy.yaml
+
+    # Install your base configuration in a Google Cloud secret
+    gcloud secrets create --data-file config.yaml pomerium-config --replication-policy automatic
+
+    # Grant the default compute account access to the secret
+    gcloud secrets add-iam-policy-binding pomerium-config \
+        --member=serviceAccount:${PROJECT}-compute@developer.gserviceaccount.com \
+        --role=roles/secretmanager.secretAccessor
+
+    # Deploy Pomerium with policy and configuration references
+    gcloud run deploy pomerium --region us-central1 --platform managed --allow-unauthenticated --max-instances 1 \
+        --image=gcr.io/pomerium-io/pomerium:v0.10.0-rc2-cloudrun \
+        --set-env-vars VALS_FILES="/pomerium/config.yaml:ref+gcpsecrets://${PROJECT}/pomerium-config",POLICY=$(base64 policy.yaml)
```

Comment:
fwiw you can separate different envs into different args (you can repeat this arg)

Reply:
👍